pkg/lib/cockpit-components-listing.jsx

```diff
@@ -253,21 +255,30 @@ export class ListingRow extends React.Component {
                 );
             }
 
+            let simpleBody, heading;
+            if ('simpleBody' in this.props) {
+                simpleBody = (
+                    <div className="listing-ct-body" key="simplebody">{this.props.simpleBody}</div>
+                );
+            } else {
+                heading = (<div className="listing-ct-head">
+                    <div className="listing-ct-actions">
+                        {listingDetail}
+                        {this.props.listingActions}
+                    </div>
+                    <ul className="nav nav-tabs nav-tabs-pf">
+                        {links}
+                    </ul>
+                </div>);
+            }
+
             return (
                 <tbody className="open">
                     {listingItem}
                     <tr className="listing-ct-panel">
                         <td colSpan={ headerEntries.length + (expandToggle ? 1 : 0) + (this.props.addCheckbox ? 1 : 0) }>
-                            <div className="listing-ct-head">
-                                <div className="listing-ct-actions">
-                                    {listingDetail}
-                                    {this.props.listingActions}
-                                </div>
-                                <ul className="nav nav-tabs nav-tabs-pf">
-                                    {links}
-                                </ul>
-                            </div>
-                            {tabs}
+                            { tabs && heading }
```

Comment:
this `tabs` just tests if tabs is defined and if is, then it shows heading? Wouldn't it make sense to just show heading when defined? (`{ heading }`)
Or wouldn something like `simpleBody || (heading && tabs)` work? (reading like "set simpleBody or heading followed by tabs")